okhttp/src/main/java/okhttp3/HttpUrl.java

```diff
@@ -772,6 +772,23 @@ public Builder addPathSegment(String pathSegment) {
       return this;
     }
 
+    public Builder addPathSegments(String groupedPathSegments) {
+      if (groupedPathSegments == null) {
+        throw new IllegalArgumentException("groupedPathSegments == null");
+      }
+      int startIndex = 0;
+      int endIndex = groupedPathSegments.indexOf('/');
+      while (endIndex != -1) {
```

Comment:
(starting with `/` means there’s going to be an empty segment in the output, like `http://example.com//x`, which is weird but legal)